### PR TITLE
Add kernelci-pipeline submodule

### PR DESCRIPTION
**Pull request by JenySadadia**
Closes https://github.com/kernelci/kernelci-project/issues/338

Add `kernelci-pipeline` as a submodule to be able to include its documentation to `kernelci.org/external`.

### CODE REVIEW

**Comment by JenySadadia** [comment]
Tested OK locally in conjunction with the below PRs:
- https://github.com/kernelci/kernelci-api/pull/533
- https://github.com/kernelci/kernelci-pipeline/pull/620

Screenshot from 2024-05-29 14-38-34

File: kernelci.org/content/en/docs/api_pipeline/_index.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+pipeline.  It also provides some [interactive API
+documentation](https://staging.kernelci.org:9000/latest/docs).  This instance
+is not stable, it's redeployed periodically with all open pull requests from
+GitHub merged together on a test integration branch.
+
+### Early Access
+
+In preparation for a full production roll-out, an [Early
+Access](/docs/api_pipeline/api/early-access) instance has been deployed in the Cloud (AKS)
+on `kernelci-api.eastus.cloudapp.azure.com`.  This is stable enough to let
```
Comment (pawiecz) [inline kernelci.org/content/en/docs/api_pipeline/_index.md:95]: Thanks for relinking Early Access guide! Let's update instance links, too (these [were changed last month](https://github.com/kernelci/kernelci-deploy/commit/509dc39e9ea7db398ab82efd38fe2110b7fe0874#diff-18ad5a095d634e9316428cbebe628c23d55a6d08bd2d29aa127eb757e15528e7))

```suggestion
on `kernelci-api.westus3.cloudapp.azure.com`.  This is stable enough to let
```

File: kernelci.org/content/en/docs/api_pipeline/_index.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+users give it a try as some form of beta-testing and is used as a candidate
+solution for an initial production deployment in the coming months.  Like
+staging, it has an auto-generated [interactive API
+documentation](https://kernelci-api.eastus.cloudapp.azure.com/latest/docs).
```
Comment (pawiecz) [inline kernelci.org/content/en/docs/api_pipeline/_index.md:99]:
```suggestion
documentation](https://kernelci-api.westus3.cloudapp.azure.com/latest/docs).
```

**Review by pawiecz — CHANGES_REQUESTED** [review]

File: kernelci.org/content/en/docs/api_pipeline/_index.md (hunk 1 above)
Comment (JenySadadia) [reply kernelci.org/content/en/docs/api_pipeline/_index.md]: Fixed it. Thanks.

File: kernelci.org/content/en/docs/api_pipeline/_index.md (hunk 2 above)
Comment (JenySadadia) [reply kernelci.org/content/en/docs/api_pipeline/_index.md]: Fixed it too.

**Review by pawiecz — APPROVED** [review]
LGTM

**Comment by JenySadadia** [comment]
I am not able to verify if the internal link works.
I'll merge this after it's verified.
@nuclearcat When will we be able to deploy this on staging? The local setup is kind of annoying me now.

**Pull request merged**